queen-infra-db/src/main/java/fr/insee/queen/infrastructure/db/surveyunit/repository/SurveyUnitDao.java

```diff
@@ -216,4 +223,41 @@ public List<SurveyUnit> findAll() {
     public void cleanExtractedData(String campaignId, Long startTimestamp, Long endTimestamp) {
         dataRepository.cleanExtractedData(campaignId, startTimestamp, endTimestamp);
     }
+
+    private void save(String surveyUnitId, String campaignId, String questionnaireId) {
+        Map<String, Object> fieldsToUpdate = new LinkedHashMap<>();
+        if (campaignId != null) {
+            fieldsToUpdate.put("campaign_id", campaignId);
+        }
+
+        if (questionnaireId != null) {
+            fieldsToUpdate.put("questionnaire_model_id", questionnaireId);
+        }
+
+        if (fieldsToUpdate.isEmpty()) {
+            return;
+        }
+
+        StringBuilder sql = new StringBuilder("UPDATE survey_unit SET ");
+
+        boolean firstKey = true;
+        for (String fieldKey : fieldsToUpdate.keySet()) {
+            if (!firstKey) {
+                sql.append(", ");
+            }
+            sql
+                .append(fieldKey)
+                .append(" = :")
+                .append(fieldKey);
+            firstKey = false;
+        }
```

Comment:
```suggestion
       String fields = fieldsToUpdate.keySet()
    .stream()
    .map(key -> key + " = :" + key)
    .collect(Collectors.joining(", "));
    String sql = "UPDATE survey_unit SET " + fields + " WHERE id = :surveyUnitId"; // could String.format here to prevent concatenation
```